### PR TITLE
Add Logs to the connection object

### PR DESCRIPTION
**Pull request by GabrielFleischer**
In the last meeting, we had issues receiving messages from the server.

The problem is that there are not enough logs related to the connection object.

This tiny PR adds **a lot** of logs in the connection object. That way, we will be able to pinpoint the cause with ease.

### CODE REVIEW

**Review by Warhedgehog — DISMISSED** [review]
Good addition 👍
Just wondering if it might make more sense to log those in info rather than debug, because it will add quite some logs I imagine

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/Connection.java [hunk 1]
```diff
@@ -1,31 +1,65 @@
...
     // subject
-    this.disposable =
-        laoService.observeMessage().subscribe(messagesSubject::onNext, messagesSubject::onError);
+    this.disposables.add(
+        laoService
+            .observeMessage()
+            .doOnNext(msg -> Log.d(TAG, "Received a new message from remote: " + msg))
+            .subscribe(messagesSubject::onNext, messagesSubject::onError));
+
+    // Add logs on connection stated
```
Comment (jbsv) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/Connection.java:36]: typo? 
```suggestion
    // Add logs on started connection
```

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/Connection.java [hunk 2]
```diff
@@ -39,7 +73,8 @@ public Observable<WebSocket.Event> observeConnectionEvents() {
   }
 
   public void close() {
-    disposable.dispose();
+    disposables.dispose();
+    messagesSubject.onComplete();
```
Comment (jbsv) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/Connection.java:77]: please add a short comment for its purpose

**Review by jbsv — CHANGES_REQUESTED** [review]

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/Connection.java (hunk 2 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/Connection.java:78]: added

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/Connection.java (hunk 1 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/Connection.java]: fixed

**Review by jbsv — APPROVED** [review]

**Comment by pierluca** [comment]
I've requested a re-review as the CI seems to fail to unrelated changes and the issues seem to have been addressed.

**Review by Warhedgehog — APPROVED** [review]
LGTM !

**Pull request merged**